components/core/src/clp/BufferedFileReader.hpp

```diff
@@ -248,14 +227,13 @@ class BufferedFileReader : public ReaderInterface {
     static constexpr size_t cDefaultBufferSize = (16 * cMinBufferSize);
 
     // Variables
-    int m_fd{-1};
-    std::string m_path;
     size_t m_file_pos{0};
+    std::unique_ptr<ReaderInterface> m_file_reader;
 
     // Buffer specific data
     std::vector<char> m_buffer;
     size_t m_base_buffer_size;
-    std::optional<BufferReader> m_buffer_reader;
+    std::unique_ptr<BufferReader> m_buffer_reader;
```

Comment:
If we modified BufferReader to support init with `data=nullptr` and `data_size=0`, then we could:

* use a default `BufferReader` initially and set it in the `BufferedFileReader` constructor.
* use copy-initialization when setting `BufferReader rather recreating it.

Any argument against doing that?

Reply:
Done